src/Tests/Microsoft.NET.TestFramework/TestAssetsManager.cs

```diff
@@ -139,7 +139,14 @@ public string GetAndValidateTestProjectDirectory(string testProjectName, string
                 }
             }
 
-            return Path.Combine(baseDirectory, directoryName.ToString());
+            var directoryPath = Path.Combine(baseDirectory, directoryName.ToString());
+
+            if (Directory.Exists(directoryPath))
+            {
+                throw new Exception($"Test dir {directoryPath} already exists");
+            }
```

Comment:
By default, we should only error out when running in CI.  Otherwise rerunning tests locally will fail by default unless you delete the test directory manually each time.

Reply:
Do we have a flag for if the tests are running in CI already?

Reply:
I think there is an MSBuild property or environment variable.  Try downloading a binlog from a CI run.  Also when the tests are run in helix there may be something that tells us about that.  @wli3

Reply:
I think we can make it an error on default. And add an extra "set environment" and a new env in .\artifacts\sdk-build-env.bat and .\artifacts\sdk-build-env.sh

Reply:
we need to run .\artifacts\sdk-build-env.bat to build sdk anyway

Reply:
> we need to run .\artifacts\sdk-build-env.bat to build sdk anyway

I basically always run it but a lot of the time it's not necessary.  I imagine most people that try to use our repo will just open the .sln in Visual Studio and they'll want the tests to work.

Reply:
Agreed, I'm went ahead and conditioned it based on the `ContinuousIntegrationBuild` property.

Reply:
>  will just open the .sln in Visual Studio and they'll want the tests to work.

nothing will work by open .sln in visual studio